deps/rabbitmq_stomp/src/rabbit_stomp_processor.erl

```diff
@@ -803,65 +1075,104 @@ negotiate_version(Frame) ->
     rabbit_stomp_util:negotiate_version(ClientVers, ?SUPPORTED_VERSIONS).
 
 
-send_delivery(Delivery = #'basic.deliver'{consumer_tag = ConsumerTag},
+deliver_to_client(ConsumerTag, Ack, Msgs, State) ->
+    lists:foldl(fun(Msg, S) ->
+                       deliver_one_to_client(ConsumerTag, Ack, Msg, S)
+                end, State, Msgs).
+
+deliver_one_to_client(ConsumerTag, _Ack, {QName, QPid, MsgId, Redelivered, MsgCont0} = _Msg,
+                      State = #proc_state{queue_states = QStates,
+                                          delivery_tag = DeliveryTag}) ->
+
+    [RoutingKey | _] = mc:get_annotation(routing_keys, MsgCont0),
+    ExchangeNameBin = mc:get_annotation(exchange, MsgCont0),
+    MsgCont = mc:convert(mc_amqpl, MsgCont0),
+    Content = mc:protocol_state(MsgCont),
+    Delivery = #'basic.deliver'{consumer_tag = ConsumerTag,
+                                delivery_tag = DeliveryTag,
+                                redelivered  = Redelivered,
+                                exchange     = ExchangeNameBin,
+                                routing_key  = RoutingKey},
```

Comment:
Creating an AMQP 0.9.1 `#basic.deliver{}` is unnecessary. The STOMP protocol is used to send the message to the client, not the AMQP 0.9.1 protocol. I think you should restructure the code, specifically `rabbit_stomp_util:headers_extra/4` should not accept a `#basic.deliver{}` anymore.